jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/editor/LogStreamTable.kt

```diff
@@ -0,0 +1,90 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.cloudwatch.logs.editor
+
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.Disposer
+import com.intellij.ui.ScrollPaneFactory
+import com.intellij.ui.table.TableView
+import com.intellij.util.ui.ListTableModel
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.launch
+import software.aws.toolkits.jetbrains.services.cloudwatch.logs.LogStreamActor
+import software.aws.toolkits.jetbrains.services.cloudwatch.logs.LogStreamEntry
+import software.aws.toolkits.jetbrains.services.cloudwatch.logs.LogStreamFilterActor
+import software.aws.toolkits.jetbrains.services.cloudwatch.logs.LogStreamListActor
+import software.aws.toolkits.jetbrains.utils.ApplicationThreadPoolScope
+import software.aws.toolkits.resources.message
+import javax.swing.JScrollBar
+import javax.swing.JScrollPane
+import javax.swing.JTable
+import javax.swing.SortOrder
+
+class LogStreamTable(
+    val project: Project,
+    logGroup: String,
+    logStream: String,
+    type: TableType
+) :
+    CoroutineScope by ApplicationThreadPoolScope("LogStreamTable"), Disposable {
+
+    enum class TableType {
+        LIST,
+        FILTER
+    }
+
+    val component: JScrollPane
```

Comment:
Can this just be a JComponent so public API isnt tied to ScollPane

Reply:
Sure, that seems reasonable